pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +180,82 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
+	existingDeployments, err := c.deploymentClient.listDeploymentsForConfig(config.Namespace, config.Name)
+	if err != nil {
+		return "", nil, false, fmt.Errorf("couldn't list deployments for deploymentConfig %s: %v", deployutil.LabelForDeploymentConfig(config), err)
+	}
+	// check if the latest deployment exists
+	// we'll return after we've dealt with the multiple-active-deployments case
+	latestDeploymentExists, latestDeploymentStatus := deployutil.LatestDeploymentInfo(config, existingDeployments)
+	if latestDeploymentExists && latestDeploymentStatus == deployapi.DeploymentStatusComplete {
+		// If the latest deployment exists and is complete, clear the dc message
+		return "", existingDeployments, latestDeploymentExists, nil
+	}
+
+	// Rest of the code will handle non-existing or failed latest deployment causes
+	// TODO: Inspect pod logs in case of failed latest
+
+	// If the configuration we are inspecting doesn't have a config change trigger
+	// there is no reason in proceeding to inspect its image change trigger
+	if !deployutil.HasChangeTrigger(config) {
+		return "", existingDeployments, latestDeploymentExists, nil
+	}
+
+	details := ""
+	// Look into image change triggers and find out possible deployment failures such as
+	// missing image stream tags with or without build configurations pointing at them
+	for _, trigger := range config.Triggers {
+		if trigger.Type != deployapi.DeploymentTriggerOnImageChange || trigger.ImageChangeParams == nil || !trigger.ImageChangeParams.Automatic {
+			continue
+		}
+		name := trigger.ImageChangeParams.From.Name
+		tag := trigger.ImageChangeParams.Tag
+
+		// Check if the image stream tag pointed by the trigger exists
+		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
+			if !errors.IsNotFound(err) {
+				return details, existingDeployments, latestDeploymentExists, err
+			}
+			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
+			// (a build configuration output points to it so it's going to be populated at some point in the
+			// future)
+			details = fmt.Sprintf("The config trigger for %q will have no effect because \"%s:%s\" does not exist", config.Name, name, tag)
+			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+			if err != nil {
+				return details, existingDeployments, latestDeploymentExists, err
+			}
+
+			for _, bc := range bcList.Items {
+				if bc.Spec.Output.To != nil && bc.Spec.Output.To.Kind == "ImageStreamTag" {
+					parts := strings.Split(bc.Spec.Output.To.Name, ":")
+					if len(parts) != 2 {
+						return details, existingDeployments, latestDeploymentExists, fmt.Errorf("invalid image stream tag: %s", bc.Spec.Output.To.Name)
+					}
+					if parts[0] == name && parts[1] == tag {
+						details = fmt.Sprintf("\"%s:%s\" does not exist yet but will be created at some point since buildConfig %q points to it", name, tag, bc.Name)
```

Comment:
How about making this a little more descriptive like the other warning... like, "The image trigger for <tag> will have no effect because <tag> doesn't exist. If a <tag> is expected, check the <name> build which produces <tag>."

Or something